src/System.Diagnostics.Process/tests/ProcessStreamReadTests.netcoreapp2.1.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Diagnostics.Tests
+{
+    public partial class ProcessStreamReadTests : ProcessTestBase
+    {
+        [Fact]
+        public void TestCustomStandardInputEncoding()
+        {
+            var process = CreateProcessPortable(RemotelyInvokable.ReadLineWithCustomEncodingWriteLineWithUtf8, Encoding.UTF32.WebName);
+            process.StartInfo.RedirectStandardInput = true;
+            process.StartInfo.StandardInputEncoding = Encoding.UTF32;
+            process.StartInfo.RedirectStandardOutput = true;
+            process.StartInfo.StandardOutputEncoding = Encoding.UTF8;
+            process.Start();
+
+            const string TestLine = "\U0001f627\U0001f62e\U0001f62f";
```

Comment:
Can you add a negative test, where SIE is set to some other encoding, and the output does not match the expected value?